### PR TITLE
Add negative tests for inline asm

### PR DESCRIPTION
**Pull request by fadeeval**
Add tests for verification of error messages for inline assembly feature.

Signed-off-by: Aleksander Fadeev aleksander.fadeev@intel.com

### CODE REVIEW

File: sycl/test/inline-asm/negative_tests/asm_unsup_instruction_divm_skl.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// UNSUPPORTED: cuda
+// REQUIRES: gpu,linux
+// RUN: %clangxx -fsycl %s -DINLINE_ASM -o %t.out
+// RUN: not %t.out 2>&1 | FileCheck %s
+
+// CHECK: Build succeeded
```
Comment (AlexeySachkov) [inline sycl/test/inline-asm/negative_tests/asm_unsup_instruction_divm_skl.cpp:6]: I wonder why do we have tests that expected to be successfully built in `negative_tests` directory?

If these tests require particular HW to catch device program build error, than I suggest to remove them as they are too fragile, IMHO

**Review by smaslov-intel — APPROVED** [review]
LGTM, but I am not familiar with GPU asm. BTW, where it is specified which GPU HW to run on?

File: sycl/test/inline-asm/negative_tests/asm_bad_opcode.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// UNSUPPORTED: cuda
+// REQUIRES: gpu,linux
+// RUN: %clangxx -fsycl %s -DINLINE_ASM -o %t.out
```
Comment (smaslov-intel) [inline sycl/test/inline-asm/negative_tests/asm_bad_opcode.cpp:3]: Why do you need this INLIONE_ASM macro since you seem to always define it?

Comment (fadeeval) [reply sycl/test/inline-asm/negative_tests/asm_bad_opcode.cpp:3]: It remained from positive tests, I will delete it.

**Review by smaslov-intel — DISMISSED** [review]
LGTM

**Comment by fadeeval** [comment]
Don't merge please yet.

**Comment by fadeeval** [comment]
I won't fix clang formatting on purpose.

**Comment by fadeeval** [comment]
The PR can be merged now

**Review by AlexeySachkov — APPROVED** [review]

**Pull request merged**